src/qt/transactionrecord.cpp

```diff
@@ -246,11 +253,16 @@ bool TransactionRecord::decomposeDebitTransaction(const CWallet* wallet, const C
                                                   const CAmount& nDebit, bool involvesWatchAddress,
                                                   QList<TransactionRecord>& parts)
 {
+    // Return early if there are no outputs.
+    if (wtx.vout.empty()) {
+        return false;
+    }
+
     CAmount nTxFee = nDebit - wtx.GetValueOut();
+    TransactionRecord sub(wtx.GetHash(), wtx.GetTxTime(), wtx.GetTotalSize());
```

Comment:
(cbf9c7c32cd8618d56194f7cb1bbda7f3a983deb) shouldn't we keep initializing a new sub-record for every output?
otherwise things like `address`, `debit`, `credit`, and `idx`, are not reset and might be carried over from a previous iteration.

Reply:
good catch. Ideally, all of them are set on the loop cycle.
Going to cache the total tx size and generate a new `TransactionRecord` per cycle to be totally sure.